evalml/pipelines/component_graph.py

```diff
@@ -319,13 +322,13 @@ def _consolidate_inputs(x_inputs, y_input, X, y):
             ww.DataTable, ww.DataColumn: The X and y transformed values to evaluate a component with
         """
         if len(x_inputs) == 0:
-            return_x = X
+            return_x = X.to_dataframe()
         else:
             return_x = pd.concat(x_inputs, axis=1)
         return_y = y
         if y_input is not None:
             return_y = y_input
-        return_x = infer_feature_types(return_x)
+        return_x = _retain_custom_types_and_initalize_woodwork(X, return_x)
```

Comment:
This method `_consolidate_inputs` currently only gets called inside `_compute_features`. If you look at the call site, you'll see the `x_inputs` is always a list of pandas dataframes. Therefore, after concatenating those pandas dataframes representing input features from the parent components in the graph, we need to call `_retain_custom_types_and_initalize_woodwork` to preserve any columns which appeared in the original woodwork input.

Reply:
I added `test_component_graph_types_multi_input_1` to check this is working properly.

Reply:
Thanks for this @freddyaboulton . I'll try to file something for this before we merge this PR. Its a great point, and yep something we'll run into eventually.